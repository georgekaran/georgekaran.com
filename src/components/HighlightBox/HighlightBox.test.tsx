import React from 'react'
import { screen } from '@testing-library/react'

import { HighlightBox } from '.'
import { renderWithTheme } from '@/test/helpers'

describe('<HighlightBox />', () => {
  it('should render with initial state', () => {
    renderWithTheme(<HighlightBox primaryText="10" secondaryText="any_text" />)
    expect(screen.getByRole('heading', { name: /10/i })).toBeInTheDocument()
    expect(screen.getByRole('heading', { name: /any_text/i })).toBeInTheDocument()
    expect(screen.getByTestId('ellipse')).toHaveStyle({
      backgroundImage: 'url("data:image/svg+xml,%3csvg width=\'100%25\' height=\'100%25\' xmlns=\'http://www.w3.org/2000/svg\'%3e%3crect width=\'100%25\' height=\'100%25\' fill=\'none\' rx=\'100\' ry=\'100\' stroke=\'%2328F5BEFF\' stroke-width=\'2\' stroke-dasharray=\'8\' stroke-dashoffset=\'71\' stroke-linecap=\'butt\'/%3e%3c/svg%3e")',
      borderRadius: '100px',
      width: '8rem',
      height: '8rem'
    })
  })

  it('should match snapshot', () => {
    const { container } = renderWithTheme(<HighlightBox primaryText="10" secondaryText="any_text" />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
