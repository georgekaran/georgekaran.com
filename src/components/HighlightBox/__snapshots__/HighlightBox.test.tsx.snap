// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['<HighlightBox /> should match snapshot 1'] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-image: url("data:image/svg+xml,%3csvg width='100%25' height='100%25' xmlns='http://www.w3.org/2000/svg'%3e%3crect width='100%25' height='100%25' fill='none' rx='100' ry='100' stroke='%2328F5BEFF' stroke-width='2' stroke-dasharray='8' stroke-dashoffset='71' stroke-linecap='butt'/%3e%3c/svg%3e");
  border-radius: 100px;
  width: 8rem;
  height: 8rem;
  margin-right: 1.6rem;
  -webkit-transition: all 0.5s ease;
  transition: all 0.5s ease;
}

.c1:hover {
  background: linear-gradient(83.89deg,#28F5BE 2.81%,#3B37FF 97.94%);
  -webkit-transform: rotate(360deg);
  -ms-transform: rotate(360deg);
  transform: rotate(360deg);
}

.c1:hover .c2 {
  color: #FFFFFF;
  -webkit-transform: scale(1.2);
  -ms-transform: scale(1.2);
  transform: scale(1.2);
}

.c3 {
  color: #1BE0AB;
  font-size: 3.2rem;
  font-weight: 600;
  -webkit-transition: all 0.5s ease;
  transition: all 0.5s ease;
}

.c4 {
  color: #FFFFFF;
  font-size: 1.6rem;
  font-weight: 300;
}

@media (max-width:768px) {
  .c1 {
    width: 6rem;
    height: 6rem;
  }
}

@media (max-width:768px) {
  .c3 {
    font-size: 2.4rem;
  }
}

@media (max-width:768px) {
  .c4 {
    font-size: 1.4rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
    data-testid="ellipse"
  >
    <h2
      class="c2 c3"
    >
      10
    </h2>
  </div>
  <h5
    class="c4"
  >
    any_text
  </h5>
</div>
`
